aderyn_core/src/detect/low/unused_imports.rs

```diff
@@ -0,0 +1,202 @@
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::error::Error;
+
+use crate::ast::{ImportDirective, NodeID};
+
+use crate::capture;
+use crate::context::browser::{
+    ExtractIdentifiers, ExtractImportDirectives, ExtractInheritanceSpecifiers,
+};
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct UnusedImportsDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+// Standalone helper function
+fn process_imports<'a>(
+    distant_relatives: &'a mut Vec<NodeID>,
+    import: &'a ImportDirective,
+    context: &'a WorkspaceContext,
+) {
+    distant_relatives.push(import.source_unit);
+    let import_source_unit = context
+        .source_units()
+        .into_iter()
+        .find(|source_unit| source_unit.id == import.source_unit);
+    match import_source_unit {
+        Some(import_source_unit) => {
+            let next_level_imports = ExtractImportDirectives::from(import_source_unit).extracted;
+            next_level_imports.iter().for_each(|next_level_import| {
+                process_imports(distant_relatives, next_level_import, context);
+            });
+        }
+        None => {
+            distant_relatives.push(-99);
+        }
+    }
+}
+
+impl IssueDetector for UnusedImportsDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
```

Comment:
@TilakMaddy Can you review this and let me know what you think? I've added some long comments to try to describe what's going on here.

Reply:
Surely will look into it : )